libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2477,6 +2477,7 @@ void TypeChecker::endVisit(NewExpression const& _newExpression)
 	solAssert(!!type, "Type name not resolved.");
 
 	_newExpression.annotation().isConstant = false;
+	_newExpression.annotation().isLValue = false;
```

Comment:
We could add an AST visitor to the soltest that checks that they are set for all nodes?

Reply:
Maybe not a bad idea - it would run on all succeeding syntax tests and indeed just check that all such annotations are set on all nodes - that should be rather quick to write, shouldn't it? And even if we don't merge it in the end, just running it once locally would be good...

Reply:
I am doing this currently, but it uncovers quite a few more cases so I suggest we merge this in the mean time while I work on enhancing this

Reply:
I'm just wondering: there may be some cases, in which it's fine not to set an annotation, where we could guarantee that it's never read... in some cases that may be better than writing non-sense default values, but it'd depend on the case (and not sure there will be such cases, just saying that we also shouldn't just blindly write default values everywhere).